PYME/Analysis/points/spherical_harmonics.py

```diff
@@ -293,8 +294,10 @@ class ScaledShell(object):
     ]
 
     def __init__(self, sampling_fraction=1.):
-        self.sampling_fraction = sampling_fraction
+        from PYME.IO import MetaDataHandler
+        self.mdh = MetaDataHandler.NestedClassMDHandler()
```

Comment:
I'm not really happy with a class having a `.mdh` attribute which is not metadata in the classical, PYME, sense. The only real overlap with the canonical metadata is that it's a key-value set of properties / settings. They are somewhat orthogonal to standard PYME metadata, and do not follow standard metadata formatting rules. 

I think I overlooked it in the original PR, where it was constrained to the .hdf output out of expediency (you effectively have a single use, custom file format), but when you might end up with an object sitting in a recipe namespace with a `.mdh` attribute, it should really be canonical metadata, not something else.